connection/cluster/ClusterClient.java

```diff
@@ -285,6 +293,8 @@ private ClusterDatabase.Replica seekPrimaryReplica() {
                 ClusterDatabase clusterDatabase = fetchDatabaseReplicas();
                 if (clusterDatabase.primaryReplica().isPresent()) {
                     return clusterDatabase.primaryReplica().get();
+                } else if (isNonExistentRaftClusterInfo(clusterDatabase)) {
+                    throw new TypeDBClientException(DB_DOES_NOT_EXIST, this.database);
```

Comment:
If any of the nodes does not have a replicator, the database does not exist.

In the cluster code: The typedb database is created only when the a database `CREATE` record is committed to the raft log. This is appended to raft log only in ClusterDatabaseManager.create only if creating the replicator succeeded on all nodes. ClusterDatabaseManager.create which is only called through TypeDBService - Hence run on the node which is processing the client request.